internal/commands/user/delete.go

```diff
@@ -0,0 +1,169 @@
+package user
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/spf13/pflag"
+)
+
+// Command is the `user delete` command
+// var Command = cli.CommandDefinition{
+// 	Use:         "delete",
+// 	Description: "Delete the user(s) from a Realm application",
+// 	Help:        "user delete",
+// 	Command:     &command{},
+// }
+
+// CommandDelete is the `user delete` command
+type CommandDelete struct {
+	inputs      deleteInputs
+	outputs     []deleteOutput
+	realmClient realm.Client
+}
+
+type deleteOutput struct {
+	user realm.User
+	err  error
+}
+
+type deleteInputs struct {
+	app.ProjectInputs
+	usersInputs
+}
+
+func (i *deleteInputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUser, flagUserShort, []string{}, flagUserDeleteUsage)
+	fs.VarP(
+		flags.NewEnumSet(&cmd.inputs.ProviderTypes, validProviderTypes),
+		flagProvider,
+		flagProviderShort,
+		flagProviderUsage,
+	)
+	fs.VarP(&cmd.inputs.State, flagState, flagStateShort, flagStateUsage)
+	fs.VarP(&cmd.inputs.Status, flagStatus, flagStatusShort, flagStatusUsage)
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Setup is the command setup
+func (cmd *CommandDelete) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile)
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, err := app.Resolve(ui, cmd.realmClient, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	users, err := cmd.inputs.ResolveUsers(ui, cmd.realmClient, app)
+	if err != nil {
+		return err
+	}
+
+	for _, user := range users {
+		err = cmd.realmClient.DeleteUser(app.GroupID, app.ID, user.ID)
+		cmd.outputs = append(cmd.outputs, deleteOutput{user: user, err: err})
+	}
+	return nil
+}
+
+// Feedback is the command feedback
+func (cmd *CommandDelete) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.outputs) == 0 {
+		return ui.Print(terminal.NewTextLog("No users to delete"))
+	}
+
+	var outputByProviderType = make(map[string][]deleteOutput)
+	for _, output := range cmd.outputs {
+		for _, identity := range output.user.Identities {
+			outputByProviderType[identity.ProviderType] = append(outputByProviderType[identity.ProviderType], output)
+		}
+	}
+	var logs []terminal.Log
```

Comment:
we can preallocate this slice since we know the length of it. 
`make([]terminal.Log, len(outputByProviderType))`